aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -2,84 +2,263 @@
+                for (; i < size - size % WIDTH; i += WIDTH) {              \
+                  scalar_t buffer[WIDTH];                                  \
+                  for (int64_t j = 0; j < WIDTH; j++)                      \
+                    buffer[j] = y[stridey * (j + i)];                      \
+                  vec256::map_(                                            \
+                      [](const Vec& x) { return vec256::op(x); },          \
+                      buffer,                                              \
+                      WIDTH);                                              \
+                  for (int64_t j = 0; j < WIDTH; j++)                      \
+                    x[stridex * (j + i)] = buffer[j];                      \
+                }                                                          \
+              }                                                            \
+              for (; i < size; i++) {                                      \
+                ZEROUPPER                                                  \
+                x[stridex * i] = opfn(y[stridey * i]);                     \
+              }                                                            \
+            }                                                              \
+          });                                                              \
+    });                                                                    \
+  }                                                                        \
+  REGISTER_DISPATCH(op##Impl, &op##_kernel)
+
+#define IMPLEMENT_KERNEL(types, op, opfn)                       \
+  static void op##_kernel(Tensor& result, const Tensor& self) { \
+    AT_DISPATCH_##types##_TYPES(self.type(), #op, [&] {         \
+      using Vec = vec256::Vec256<scalar_t>;                     \
+      _parallel_vector_map<scalar_t>(                           \
+          [](const Vec& x) { return vec256::op(x); },           \
+          [](const scalar_t y) {                                \
+            ZEROUPPER                                           \
+            return opfn(y);                                     \
+          },                                                    \
+          result,                                               \
+          self);                                                \
+    });                                                         \
+  }                                                             \
+  REGISTER_DISPATCH(op##Impl, &op##_kernel)
+
+#define IMPLEMENT_KERNEL_LOOP(types, op, opfn)                  \
+  static void op##_kernel(Tensor& result, const Tensor& self) { \
+    AT_DISPATCH_##types##_TYPES(self.type(), #op, [&] {         \
+      CPU_tensor_parallel_apply2<scalar_t, scalar_t>(           \
+          result, self, [](scalar_t& x, scalar_t& y) {          \
+            ZEROUPPER                                           \
```

Comment:
Why do we ZEROUPPER before every call if we're only using those calls? It seems unnecessary